src/core/layerobserver.cpp

```diff
@@ -296,7 +296,7 @@ void LayerObserver::addLayerListeners()
       if ( mObservedLayerIds.contains( vl->id() ) )
         continue;
 
-      if ( !vl->readOnly() && QFieldCloudUtils::isCloudAction( vl ) )
+      if ( !vl->readOnly() )
```

Comment:
Hey @suricactus , am I missing something here or that change doesn't guard from adding listeners to non-cloud projects? I.e. this slot is called when layers are added to the project instance and it doesn't check whether the current project is a cloud project AFAIK:

```
void LayerObserver::onLayersAdded( const QList<QgsMapLayer *> &layers )
{
  Q_UNUSED( layers );
  addLayerListeners();
}
```

You'd need to add a cloud project check here.

Reply:
@nirvn  I have tested the apk, and it works well. 
Do you want me to approve the pr?

Reply:
We still need to tweak code to only cover could projects

Reply:
I did download the simple_Bee_Farming_No_offline project deleted and added an apiary

Reply:
@lucienicolier  Did you also synchronize, push changes in between? Can you also test editing attributes and geometries?
Also would be good to test changing the same feature (or child features) with different users in parallel.

Reply:
It can be good to know that I was adding a collaborator and downloading the project on my phone with this new user

Reply:
I logged in with another user and made modifications to the same features as I just did, and it took 5 min to sync to the cloud. But everything went well.